mopidy/local/playlists.py

```diff
@@ -67,38 +77,53 @@ def refresh(self):
     def save(self, playlist):
         assert playlist.uri, 'Cannot save playlist without URI'
 
-        old_playlist = self.lookup(playlist.uri)
-
-        if old_playlist and playlist.name != old_playlist.name:
-            playlist = playlist.copy(name=formatting.slugify(playlist.name))
-            playlist = self._rename_m3u(playlist)
-
-        self._save_m3u(playlist)
-
-        if old_playlist is not None:
-            index = self._playlists.index(old_playlist)
+        uri = playlist.uri
+        # TODO: require existing (created) playlist - currently, this
+        # is a *should* in https://docs.mopidy.com/en/latest/api/core/
+        try:
+            index = self._playlists.index(self.lookup(uri))
+        except ValueError:
+            logger.warn('Saving playlist with new URI %s', uri)
+            index = -1
+
+        playlist = self._save_m3u(playlist)
+        if index >= 0 and uri != playlist.uri:
+            path = local_playlist_uri_to_path(uri, self._playlists_dir)
+            if os.path.exists(path):
+                os.remove(path)
```

Comment:
In this case, I'd rather escalate the error to the client, because something must have been gone terribly wrong. One use case I can think of is lack of access rights: We don't want to remove the playlist from `self._playlists` if we can't delete the file, I guess.